packages-exp/auth-exp/test/integration/webdriver/util/auth_driver.ts

```diff
@@ -46,57 +37,78 @@ export class AuthDriver {
   async start(browser: string): Promise<void> {
     await authTestServer.start();
     this.webDriver = await new Builder().forBrowser(browser).build();
-    await this.webDriver.get(authTestServer.address!);
-    await this.injectConfigAndInitAuth();
+    await this.goToTestPage();
   }
 
   async stop(): Promise<void> {
     authTestServer.stop();
     await this.webDriver.quit();
   }
 
-  async call<T>(fn: TestFunction): Promise<T> {
+  async call<T>(fn: string, ...args: unknown[]): Promise<T> {
     // When running on firefox we can't just return result immediately. For
     // some reason, the binding ends up causing a cycle dependency issue during
     // serialization which blows up the whole thing. It's okay though; this is
     // an integration test: we don't care about the internal (hidden) values of
     // these objects.
-    const result = await this.webDriver.executeAsyncScript(`
+    const {
+      type,
+      value
+    }: {
+      type: string;
+      value: string;
+    } = await this.webDriver.executeAsyncScript(
+      `
       var callback = arguments[arguments.length - 1];
-      ${fn}().then(result => {
-        callback(JSON.stringify(result));
+      ${fn}(...arguments).then(result => {
```

Comment:
Gotta drop the last argument since that's the webDriver provided callback function (used above)